src/parquet/column/reader.h

```diff
@@ -47,6 +47,66 @@ struct hash<parquet::Encoding::type> {
 
 namespace parquet_cpp {
 
+class LevelDecoder {
+
+ private:
+  parquet::Encoding::type encoding_;
+  std::unique_ptr<RleDecoder> rle_decoder;
+  std::unique_ptr<BitReader> bit_packed_decoder;
+
+ public:
+  LevelDecoder() {}
+
+  //return true if one of the decoder is set
+  bool set() {
+     return ((NULL != rle_decoder.get()) || (NULL != bit_packed_decoder.get()));
+  }
+
+  // Initialize the LevelDecoder.
+  void Init(parquet::Encoding::type encoding, int16_t max_level, int num_buffered_values,
+             const uint8_t** data, int* data_size) {
```

Comment:
Modifying `data` and `data_size` is not a good pattern to use. Instead, `Init` should accept `const uint8_t* data` and `size_t data_size` and return `size_t` (instead of `void`), indicating the number of bytes consumed by initializing the level decoder.